gene2phenotype_project/gene2phenotype_app/serializers.py

```diff
@@ -980,111 +986,102 @@ class Meta:
+        curation_entry = self.compare_curation_data(data_dict, user_obj.id)
+   
+        if curation_entry: # Throw error if data is already stored in table
+            raise serializers.ValidationError({"message": f"Data already under curation. Please check session '{curation_entry.session_name}'"})
+        else:
+            if len(data_dict["json_data"]["panels"]) >= 1:
+                panels = UserSerializer.get_panels(self,user_obj.id)
+                panels = ['Developmental disorders' if panel == "DD" else panel for panel in panels] # turning DD to developmental disorders
+                 # Check if any panel in data_dict["json_data"]["panels"] is not in the updated panels list
+                unauthorized_panels = [panel for panel in data_dict["json_data"]["panels"] if panel.replace(' disorders', '') not in panels]
+                if unauthorized_panels:
+                    unauthorized_panels_str = "', '".join(unauthorized_panels)
+                    raise serializers.ValidationError({"message" : f"You do not have permission to curate on these panels: '{unauthorized_panels_str}'"})
 
-    # Method to create an entry in the curation data table
-    # This method is used once to create the entry, other updates are done in the endpoint to update the entry
-    # 'validated_data' contains the data already validated
-    @transaction.atomic
-    def create(self, validated_data):
-        locus = validated_data.get('locus')
-        allelic_requirement = validated_data.get('allelic_requirement')
-        disease = validated_data.get('disease')
-        date_created = datetime.now()
-        date_reviewed = date_created
-        session_name = validated_data.get('session_name')
+        return data
 
-        # In the HTML form the session_name is a list
-        # Change it to a string
-        if isinstance(session_name, list):
-            if session_name[0] == "":
-                session_name = ""
-            else:
-                session_name = session_name[0]
+    def convert_to_dict(self, data):
+        """
+            Convert data to a regular dictionary if it is an OrderedDict.
 
-        # Build the JSON
-        # We should not use the 'validated_data' because the data insert by the HTML form could have a different format
-        data_json_input = CurationDataSerializer.format_json(validated_data)
+            Parameters:
+                data: Data to convert (can be OrderedDict or dict).
 
-        # Assuming the user is already validated in the view AddCurationData
-        user_email = self.context.get('user')
-        user_obj = User.objects.get(email=user_email)
+            Returns:
+                Converted data as a dict.
+        """
+        if isinstance(data, OrderedDict):
+            return dict(data)
+        return data
+    
+    #using the Deepdiff module, compare JSON data 
+    # this still needs to be worked on when we have fixed the user permission issue 
```

Comment:
Can add a comment explaining the output.
What does it mean to return data? Are the objects the same or not.